packages/grid/_modules_/grid/GridComponent.tsx

```diff
@@ -119,6 +119,17 @@ export const GridComponent = React.forwardRef<HTMLDivElement, GridComponentProps
 
   const onResize = React.useCallback(
     (size: ElementSize) => {
+      if (process.env.NODE_ENV !== 'production') {
+        if (size.height === 0) {
+          console.error(
```

Comment:
why do we use console? How about the logger?

Reply:
From what I understand, logger is for debugging in dev and prod. This warning is dev only and should never be tempered with.

Reply:
logger can be used with consumer implementation of logging. For example Buyan, seq logging..., using console will prevent the error going to those tools.
This error is dev only as there is the if condition on the env above, not because it uses the console.
BTW why is there a different behaviour bw dev and prod?

Reply:
It's not obvious. They could be different interpretations of how to handle this case. A couple of thoughts:

1. React never console messages that aren't exceptions in production. I don't think that we should either for consistency with React and `@material-ui/core`. Removing `process.env.NODE_ENV !== 'production'` would break this constraint.
2. We could throw an error directly, however, this case isn't fatal. It could simply be coming from an intermediary state where the grid's container has no height.
3. We could use a console.warn instead of a console.error. No objections. I wasn't sure how important this should be considered. Considering the argument against throwing, console.warn could be better.
4. We never use `logger.warn` yet.
5. We only use `logger.error` once, in the ErrorBoundary. IMHO, we shouldn't have an ErrorBoundary. It's too opinionated for any shared UI components to handle. It breaks the default error forwarding flow. It's also very simple to implement a demo in the docs should be enough. So far, we decided to try it out and see the feedback we get from developers.
6. The only case I see it would make sense to use the logger here is if we want to output this warning in production too without logging it in the console (per 1. only in dev). I don't have a specific point of view on that. Why not. IMHO, 80% of the value of this warning is for configuring the grid, to reduce the onboarding friction as much as possible.

Reply:
How about we go for a logger.warn without the env check?
So if users decide to not care about it then, it will not appear in prod as the logger is noop by default in prod. They can always change this behavior.
Single point of failure.

Reply:
logger.error will not show the error message...
Only throw exception will do, to avoid having the users' app crashing

Reply:
> How about we go for a logger.warn without the env check?

So option 6? The only downside I can think of is that it retains the logic in production, accounting for bundle size. I think that if we change the logic of the default console appender to be silent in production for non error (unless forced), sure. I think it can work great.

Reply:
in prod it won't be displayed as the logger is null